### DISCUSSION TITLE
Limits in systemd service example are too strict for huge boxes

### DISCUSSION THREAD

**Discussion by jackyyf** [dani-garcia/vaultwarden#2538] [Q&A]
### Subject of the issue
Limits in systemd service example are too strict for huge boxes

### Deployment environment

* vaultwarden version: 1.25.0

* Install method: built from source

* Clients used: N/A

* Reverse proxy and version: N/A

* MySQL/MariaDB or PostgreSQL version: N/A

* Other relevant details:
The server running vaultwarden is a Dual E5-2696v3, with 36 cores and 72 threads. 

### Steps to reproduce

Use the reference systemd service file in wiki https://github.com/dani-garcia/vaultwarden/wiki/Setup-as-a-systemd-service

### Expected behaviour
Vaultwarden booted up.

### Actual behaviour
Vaultwarden panicked with some strange tracebacks, logs attached.

### Troubleshooting data
```
[2022-06-07 04:10:38.100][panic][ERROR] thread 'main' panicked at 'Cannot drop a runtime in a context where blocking is not allowed. This happens when a runtime is dropped from within an asynchronous context.': <redacted>/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.18.2/src/runtime/blocking/shutdown.rs:51
   0: vaultwarden::init_logging::{{closure}}
   1: std::panicking::rust_panic_with_hook
   2: std::panicking::begin_panic::{{closure}}
   3: std::sys_common::backtrace::__rust_end_short_backtrace
   4: std::panicking::begin_panic
   5: tokio::runtime::blocking::shutdown::Receiver::wait
   6: tokio::runtime::blocking::pool::BlockingPool::shutdown
   7: core::ptr::drop_in_place<tokio::runtime::blocking::pool::BlockingPool>
   8: core::ptr::drop_in_place<vaultwarden::schedule_jobs::{{closure}}::{{closure}}>
   9: core::ptr::drop_in_place<std::thread::Builder::spawn_unchecked_<vaultwarden::schedule_jobs::{{closure}}::{{closure}},()>::{{closure}}>
  10: std::sys::unix::thread::Thread::new
  11: std::thread::Builder::spawn
  12: vaultwarden::main::{{closure}}
  13: tokio::park::thread::CachedParkThread::block_on
  14: tokio::runtime::thread_pool::ThreadPool::block_on
  15: rocket::async_main
  16: vaultwarden::main
  17: std::sys_common::backtrace::__rust_begin_short_backtrace
  18: std::rt::lang_start::{{closure}}
  19: std::rt::lang_start_internal
  20: main
  21: __libc_start_main
  22: _start
```

### Suggestion
Limit maximum threads for pool to something like 16 or 32 (instead of the default `$(nproc)`) would be a good choice :) I don't think our nice program would have such huge load that requires a bunch of threads to handle

**Comment by BlackDex** [comment]
If you have suggestions or notes or additions too be done on the wiki you are free to add them. If you do that please also add the reason why, because all environments are different and other values could apply to other users.

**Comment by JensTimmerman** [comment]
I'm seeing the same issue on a 40 core box with `LimitNPROC=64` set in the service file, what should this be set to?
setting to 32 gives me :
Jul  6 11:22:05 bitwarden systemd[235579]: vaultwarden.service: Failed to execute /opt/bitwarden_rs/vaultwarden: Resource temporarily unavailable
(in an lxc container, the wiki recommends me to completely remove it?)
setting to 128 seems to work
